evennia/contrib/mail.py

```diff
@@ -0,0 +1,220 @@
+                        return
+                    else:
+                        self.caller.msg("Message does not exist.")
+                        return
+
+            else:
+                if self.rhs:
+                    if "/" in self.rhs:
+                        subject, body = self.rhs.split("/", 1)
+                    else:
+                        body = self.rhs
+                    self.send_mail(self.lhslist, subject, body, self.caller)
+                else:
+                    message = self.get_all_mail()[int(self.lhs) - 1]
+
+                    messageForm = []
+                    if message:
+                        messageForm.append(HEAD_CHAR * _WIDTH)
+                        messageForm.append("|wFrom:|n %s" % (message.senders[0].key))
+                        messageForm.append("|wSent:|n %s" % message.db_date_created.strftime("%m/%d/%Y %H:%M:%S"))
+                        messageForm.append("|wSubject:|n %s" % message.header)
+                        messageForm.append(SUB_HEAD_CHAR * _WIDTH)
+                        messageForm.append(message.message)
+                        messageForm.append(HEAD_CHAR * _WIDTH)
+                    self.caller.msg("\n".join(messageForm))
+                    message.tags.remove("u", category="mail")
+                    message.tags.add("o", category="mail")
+
+        else:
+            messages = self.get_all_mail()
+
+            if messages:
+                table = evtable.EvTable("|wID:|n", "|wFrom:|n", "|wSubject:|n", "|wDate:|n", "|wSta:|n",
+                                        table=None, border="header", header_line_char=SUB_HEAD_CHAR, width=_WIDTH)
+                index = 1
+                for message in messages:
+                    table.add_row(index, message.senders[0], message.header,
+                                  message.db_date_created.strftime("%m/%d/%Y"),
+                                  str(message.db_tags.last().db_key.upper()))
+                    index += 1
+
+                table.reformat_column(0, width=6)
+                table.reformat_column(1, width=17)
+                table.reformat_column(2, width=34)
+                table.reformat_column(3, width=13)
+                table.reformat_column(4, width=7)
+
+                self.caller.msg(HEAD_CHAR * _WIDTH)
+                self.caller.msg(table)
+                self.caller.msg(HEAD_CHAR * _WIDTH)
+            else:
+                self.caller.msg("Sorry, you don't have any messages.  What a pathetic loser!")
+
+    def get_all_mail(self):
+        """
+        Returns a list of all the messages where the caller is a recipient.
+        
+        Returns:
+            messages (list): list of Msg objects.
+        """
+        mail_messages = Msg.objects.get_by_tag(category="mail")
+        messages = []
+        messages = [m for m in mail_messages if self.caller.player in m.receivers]
+        return messages
+
+    def send_mail(self, recipients, subject, message, caller):
```

Comment:
As a small idea, you may want to do `Msg.objects.filter(db_receivers_players=self.caller.play).get_by_tag(category="mail")` rather than doing the `in m.receivers` check in the list comprehension - having a for loop check membership and hit the database each time could make the command run really slowly if you have a lot of objects, while a filter by player name would only hit it once. I ran into a similar situation using Msg objects for bulletin boards and checking receivers to see read posts.